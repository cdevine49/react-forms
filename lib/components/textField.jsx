import React from 'react';
import FormField from './formField';
import TextInput from './TextInput';
import Error from '../elements/error';
import Label from './label';
import Underline from '../elements/underline'
import setClassName from '../helpers/setClassName';

const TextField = ({ displayErrors, errors, required, validate, value, ...props}) => {
  return (
    <FormField
      displayErrors={ displayErrors }
      errors={ errors }
      required={ required }
      validate={ validate }
      value={ value }
    >
      { (errorMessage, displayErrors) => (
        <TextInput
          errorMessage={ errorMessage }
          onBlur={ displayErrors }
          required={ required }
          value={ value }
          { ...props }
          />
      ) }
    </FormField>
  );
}

TextField.defaultProps = {
  type: 'text',
  value: '',
  onChange: function(){},
  containerProps: {},
  errorProps: {},
  labelProps: {},
  underlineProps: {}
}

export default TextField;
