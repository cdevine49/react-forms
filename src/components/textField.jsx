import React from 'react';
import withErrors from './withErrors';
import Error from './error';
import Label from './label';
import Underline from './underline'
import setClassName from '../helpers/setClassName';

const TextField = ({
                      containerProps: {
                        className: containerClass,
                        ...containerProps
                      },
                      label,
                      labelProps,
                      id, className,
                      hideErrors,
                      displayErrors,
                      errorMessage,
                      errorProps,
                      underline,
                      underlineProps,
                      ...props
                   }) => {
  return (
    <div className={ setClassName(['input-container', containerClass]) } { ...containerProps }>
      <Label htmlFor={id} { ...labelProps }>{ label }</Label>
      <input
        id={ id }
        className={ setClassName([className, (errorMessage && 'error')]) }
        onFocus={hideErrors}
        onBlur={displayErrors}
        { ...props }
        />
      <Error { ...errorProps }>{errorMessage}</Error>
      <Underline { ...underlineProps }>{underline}</Underline>
    </div>
  );
}

TextField.defaultProps = {
  type: 'text',
  value: '',
  onChange: function(){},
  containerProps: {},
  errorProps: {},
  labelProps: {},
  underlineProps: {}
}

export default withErrors(TextField);
