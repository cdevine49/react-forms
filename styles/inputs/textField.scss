.flag-container {
  position: relative;
}

.one-line > div > .input-container {
  flex: 1;
}

.one-line {
  width: 75%;
}

.one-line > div > .input-container:not(:first-child) {
  margin-left: 10px;
}

fieldset > legend {
  font-weight: bold;
}

.one-line > div {
  display: flex;
  flex-direction: row;
}