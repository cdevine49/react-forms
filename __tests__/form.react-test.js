import React from 'react';
import Form from '../src/components/form';
import textField from '../src/components/textField';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe("Form", () => {
  describe("Default", () => {
    test("renders with default props", () => {
      const tree = renderer.create(
        <Form
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  test("accepts props", () => {
    const tree = renderer.create(
      <Form
        id="formId"
        className="new-class"
        noValidate={false}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("Submission", () => {
    let onSubmit;
    const form = mount(
      <Form>
        <input type="submit" />
      </Form>
    );

    beforeEach(() => {
      onSubmit = jest.fn();
      form.setProps({ onSubmit });
    });

    test("calls onSubmit prop when submittable array empty", () => {


      form.setState({ submittable: [] });
      expect(onSubmit.mock.calls.length).toBe(0);
      form.simulate('submit');
      expect(onSubmit.mock.calls.length).toBe(1);
    });

    test("calls onSubmit prop when submittable array elements all truthy", () => {
      form.setState({ submittable: [ true, 'true', 1] });
      expect(onSubmit.mock.calls.length).toBe(0);
      form.simulate('submit');
      expect(onSubmit.mock.calls.length).toBe(1);
    });

    test("does not call onSubmit prop when any submittable array elements falsey", () => {
      expect(onSubmit.mock.calls.length).toBe(0);
      form.setState({ submittable: [ true, 'true', false, 1] });
      form.simulate('submit');

      expect(onSubmit.mock.calls.length).toBe(0);

      form.setState({ submittable: [ true, 'true', '', 1] });
      form.simulate('submit');
      expect(onSubmit.mock.calls.length).toBe(0);
    });
  });
});