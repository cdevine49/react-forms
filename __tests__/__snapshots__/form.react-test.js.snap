// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Default renders with default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 400px;
  padding: 20px;
}

<form
  className="c0"
  noValidate={true}
  onSubmit={[Function]}
/>
`;

exports[`Form accepts props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 400px;
  padding: 20px;
}

<form
  className="new-class c0"
  id="formId"
  noValidate={false}
  onSubmit={[Function]}
/>
`;
